main.go

```diff
@@ -78,6 +76,31 @@ func main() {
 		regArr = append(regArr, re)
 	}
 
+	if len(flag.Args()) == 0 {
+		getBuf(os.Stdin, 0)
+	} else {
+		for i := range flag.Args() {
+			file, fileErr := os.Open(flag.Args()[i])
+			if fileErr != nil {
+				fmt.Fprintln(os.Stderr, fileErr)
+				os.Exit(2)
+			}
+
+			getBuf(file, i)
+			file.Close()
+		}
+	}
+
+	if counter <= 0 {
+		os.Exit(1)
+	}
+}
+
+func getBuf(file *os.File, u int) {
```

Comment:
What's u for?

Reply:
to print the file name in getBuf

Reply:
Any way to get the name w/o u?

Reply:
create a global var and set it to i in the `for i := range flag.Args() `loop i can use it in getBuf aswell but dont need to give it to the func as a parameter

Reply:
No additional vars! Get the name with all you already have, but w/o u.

Reply:
Tip: file.